qa/rpc-tests/zkey_import_export.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python2
+# Copyright (c) 2017 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from decimal import Decimal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_greater_than, start_nodes, initialize_chain_clean, connect_nodes_bi
+
+import logging
+import time
+import math
+
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+class ZkeyImportExportTest (BitcoinTestFramework):
+
+    def setup_chain(self):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 5)
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(5, self.options.tmpdir )
+        connect_nodes_bi(self.nodes,0,1)
+        connect_nodes_bi(self.nodes,1,2)
+        connect_nodes_bi(self.nodes,0,2)
+        connect_nodes_bi(self.nodes,0,3)
+        connect_nodes_bi(self.nodes,0,4)
+        self.is_network_split=False
+        self.sync_all()
+
+    # TODO: Refactor in z_addr test_framework file
+    # Returns txid if operation was a success or None
+    def wait_and_assert_operationid_status(self, node, myopid, in_status='success', in_errormsg=None):
+        print('waiting for async operation {}'.format(myopid))
+        opids = []
+        opids.append(myopid)
+        timeout = 300
+        status = None
+        errormsg = None
+        txid = None
+        for x in xrange(1, timeout):
+            results = node.z_getoperationresult(opids)
+            if len(results)==0:
+                time.sleep(1)
+            else:
+                print("Results", results[0])
+                status = results[0]["status"]
+                if status == "failed":
+                    errormsg = results[0]['error']['message']
+                elif status == "success":
+                    txid = results[0]['result']['txid']
+                break
+        print('...returned status: {}'.format(status))
+        assert_equal(in_status, status)
+        if errormsg is not None:
+            assert(in_errormsg is not None)
+            assert_equal(in_errormsg in errormsg, True)
+            print('...returned error: {}'.format(errormsg))
+        return txid
+
+    def run_test(self):
+        [alice, bob, charlie, david, miner] = self.nodes
+
+        def z_send(from_node, from_addr, to_addr, amount):
+            opid = from_node.z_sendmany(from_addr, [{"address": to_addr, "amount": Decimal(amount)}])
+            txid = self.wait_and_assert_operationid_status(from_node, opid)
+            self.sync_all()
+            miner.generate(1)
+            self.sync_all()
+
+        def z_getbalance(node, zaddr):
+            bal = node.z_getbalance(zaddr)
+            # Ignore fees for sake of comparison
+            round_balance = math.ceil(bal*100)/100
```

Comment:
The problem here is likely in the floating-point division. I've opened #2807 to ensure we address this later.

Reply:
ah that makes sense, thanks